107/src/test/java/org/ehcache/docs/EhCache107ConfigurationIntegrationDocTest.java

```diff
@@ -166,4 +184,51 @@ public void testWithoutEhcacheExplicitDependencyAndNoCodeChanges() throws Except
     assertThat(foosEhcacheConfig.getExpiry().getExpiryForCreation(42L, "Answer!"),
         is(new org.ehcache.expiry.Duration(2, TimeUnit.MINUTES)));
   }
+
+  @Test
+  public void testTemplateOverridingStoreByValue() throws Exception {
+    cacheManager = cachingProvider.getCacheManager(
+        getClass().getResource("/org/ehcache/docs/ehcache-jsr107-template-override.xml").toURI(),
+        getClass().getClassLoader());
+
+    MutableConfiguration<Long, String> mutableConfiguration = new MutableConfiguration<Long, String>();
+    mutableConfiguration.setTypes(Long.class, String.class);
+
+    Cache<Long, String> myCache = null; 
+    myCache = cacheManager.createCache("anyCache", mutableConfiguration);
+    myCache.put(1L, "foo");
+    assertNotSame("foo", myCache.get(1L));
+
+    myCache = cacheManager.createCache("byRefCache", mutableConfiguration);
+    myCache.put(1L, "foo");
+    assertSame("foo", myCache.get(1L));
+
+    myCache = cacheManager.createCache("weirdCache", mutableConfiguration);
+    myCache.put(1L, "foo");
+    assertNotSame("foo", myCache.get(1L));
```

Comment:
In reference to your previous comment, what should `myCache.getConfiguration(Configuration.class).isStoreByValue()` return for this *weirdCache* that's neither store-by-value nor store-by-ref but a mix of both?